enhancements/network/cloud-egress-ip.md

```diff
@@ -0,0 +1,445 @@
+  `iptables -t nat -D POSTROUTING -p tcp -m tcp -d $NODE_IP/32 -j
+  SNAT--to-source $ALIAS_IP` to make things work. It is believed to be a bug in
+  OVN, but further coordination with them will be required to figure out why it
+  can't SNAT, whereas that iptables rule works. This is not a concern for
+  openshift-sdn, as it programs that exact iptables rule by default for egress
+  IP.
+
+[8]:
+https://cloud.google.com/sdk/gcloud/reference/compute/instances/network-interfaces/update
+
+#### Model
+
+##### network plugin <-> cloud API controller
+
+The network plugin needs to remain "master" of the egress IP setup. The reason
+for this is that the network plugin is the one and only component which health
+checks routes to egress nodes and re-balances the egress IP assignment depending
+on the result. This component will need to execute what the network plugin has
+determined is best.
+
+Given the above, it is however important that the network plugin is conveyed the
+cloud's information concerning the nodes subnet range. This is especially
+evident on GCP, as without it: the network plugin won't be able to perform any
+assignments at all, given the nodes default `/32` subnet.
+
+##### CRD vs. a labelled / annotated approach
+
+The cloud's node subnet information needs to be communicated to the network
+plugins. Two models have been imagined for this: a CRD based approach, or a node
+labelled / annotated approach.
+
+**CRD**
+
+The cloud component could use a dedicated CRD that the network plugin watches.
+The CRD would store the nodes subnet, as known by the cloud, and perhaps contain
+a status field indicating a successful cloud assignment of the egress IP. Doing
+this would however introduce explicit dependencies between the network plugin
+and this cloud component. If the component ends up being a dedicated OpenShift
+one: the CRD would end up creating a dependency tree between upstream
+OVN-Kubernetes and OpenShift specific components, which is anything but
+desirable.
+
+**labelled / annotated approach**
+
+The second approach imagined is labelling / annotating the Kubernetes node
+object. The cloud component would query the cloud API for the nodes subnet range
+and label / annotate the node objects according to a defined convention which
+the network plugins can read. That information would then be taken into account
+during the plugin's assignment procedure.
+
+Given the minimal information needed to be conveyed to the network plugin from
+the cloud component, and the dependencies that a CRD would introduce: it is
+estimated that the labelled / annotated model might be preferred. The only thing
+not covered by that approach is indicating to the user if the cloud egress IP
+assignment was successful or not. There are other Kubernetes objects however
+that could cover that: events, for example. This model also does not provide a
+certified Kubernetes reconciliation loop, as the CRD model would.
```

Comment:
There are lots of ways to provide feedback. For example, you could provide a second annotation indicating the IPs currently assigned to a given Node.

You are right, however, that there's no Status object you could write.

Reply:
Yes, I will mention that.